imap_processing/codice/tests/test_decom.py

```diff
@@ -52,6 +54,131 @@ def validation_data():
+    #  coding them here
+    mnemonics_to_process = [
+        "SPIN_BIN_PERIOD",
+        "SPIN_PERIOD",
+        "SPIN_PERIOD_TIMER",
+        "OPTICS_HV_DAC_ESA_A",
+        "OPTICS_HV_DAC_ESA_B",
+        "OPTICS_HV_DAC_IONBULK",
+        "SENSOR_HV_DAC_SSDO",
+        "SENSOR_HV_DAC_SSDB",
+        "SENSOR_HV_DAC_APDB",
+        "SENSOR_HV_DAC_APDB2",
+        "SENSOR_HV_DAC_START_MCP",
+        "SENSOR_HV_DAC_STOP_MCP",
+        "SENSOR_HV_DAC_STOP_OPTICS_GRID",
+        "SBULK_VMON",
+        "SSDO_VMON",
+        "SSDB_VMON",
+        "APDB1_VMON",
+        "APDB2_VMON",
+        "IOBULK_VMON",
+        "ESAA_HI_VMON",
+        "SPARE_62",
+        "STRMCP_VMON",
+        "STPMCP_VMON",
+        "STPOG_VMON",
+        "APDB1_IMON",
+        "ESAB_HI_VMON",
+        "SPARE_68",
+        "APDB2_IMON",
+        "SSDB_IMON",
+        "STPMCP_IMON",
+        "IOBULK_IMON",
+        "STRMCP_IMON",
+        "MDM25P_14_T",
+        "MDM25P_15_T",
+        "MDM25P_16_T",
+        "MDM51P_27_T",
+        "IO_HVPS_T",
+        "LVPS_12V_T",
+        "LVPS_5V_T",
+        "LVPS_3P3V_T",
+        "LVPS_3P3V",
+        "LVPS_5V",
+        "LVPS_N5V",
+        "LVPS_12V",
+        "LVPS_N12V",
+        "LVPS_3P3V_I",
+        "LVPS_5V_I",
+        "LVPS_N5V_I",
+        "LVPS_12V_I",
+        "LVPS_N12V_I",
+        "CDH_1P5V",
+        "CDH_1P8V",
+        "CDH_3P3V",
+        "CDH_12V",
+        "CDH_N12V",
+        "CDH_5V",
+        "CDH_5V_ADC",
+        "CDH_PROCESSOR_T",
+        "CDH_1P8V_LDO_T",
+        "CDH_1P5V_LDO_T",
+        "CDH_SDRAM_T",
+        "SNSR_HVPS_T",
+    ]
+
+    filename_eu = "idle_export_eu.COD_NHK_20230822_122700.csv"
+    eu_csv = pd.read_csv(filename_eu)
+
+    # Filter the DataFrame to keep only columns that match the mnemonics_to process
+    filtered_eu_data = eu_csv[mnemonics_to_process]
+
+    return filtered_eu_data
+
+
+def validate_unit_conversion(validation_data, filtered_analog):
+    """Convert validation data to engineering units using the coefficients from the
+    AnalogConversions packet and compare it with filtered_eu_data.
+
+    Parameters
+    ----------
+    validation_data : pandas DataFrame
+        The validation data with raw values
+    filtered_analog : pandas DataFrame
+        The DataFrame containing coefficients for unit conversion
+
+    Returns
+    -------
+    engineering_data : pandas DataFrame
+        The DataFrame with engineering unit values
+    """
+    engineering_data = pd.DataFrame(
+        np.nan, index=validation_data.index, columns=validation_data.columns
+    )
+
+    for mnemonic in validation_data.columns:
+        matching_row = filtered_analog[filtered_analog["mnemonic"] == mnemonic]
+
+        if not matching_row.empty:
+            coefficients = matching_row[
+                ["c0", "c1", "c2", "c3", "c4", "c5", "c6", "c7"]
+            ].values[0]
+
+            raw_data = validation_data[mnemonic]
+            engineering_unit_data = (
```

Comment:
I think @tech3371 was using Numpy's Polynomial class for this, are you able to do the same?
https://numpy.org/doc/stable/reference/generated/numpy.polynomial.polynomial.Polynomial.html#numpy.polynomial.polynomial.Polynomial

`np.polynomial.Polynomial(coefficients)(raw_data)`

Reply:
@tech3371 did mention this to me in person and I saw it in her PR. I will look at her code again and look into this.